src/ray/core_worker/core_worker.cc

```diff
@@ -3160,6 +3154,12 @@ Status CoreWorker::GetAndPinArgsForExecutor(const TaskSpecification &task,
       reference_counter_->AddBorrowedObject(
           arg_id, ObjectID::Nil(), task.ArgRef(i).owner_address());
       borrowed_ids->push_back(arg_id);
+      // We need to put an OBJECT_IN_PLASMA error here so the subsequent call to Get()
+      // properly redirects to the plasma store.
+      if (!options_.is_local_mode) {
```

Comment:
Can you add a NOTE here about the ordering of the call with respect to reference counter? Seems pretty subtle.